internal/resto/client.go

```diff
@@ -99,12 +102,18 @@ func (c *Client) PollJob(ctx context.Context, id string, interval, timeout time.
 	deathclock := time.NewTimer(timeout)
 	defer deathclock.Stop()
 
+	var retryTime int
 	for {
 		select {
 		case <-ticker.C:
 			j, err = doRequest(c.HTTPClient, request)
```

Comment:
The underlying request code is 100% identical between the two clients.
Might want to invest a bit more time here to consolidate the logic. A retryable request, so to speak.
Might be worthwhile to investigate something like https://github.com/hashicorp/go-retryablehttp as well.

Reply:
Nice one!